pennylane/interfaces/batch/autograd.py

```diff
@@ -101,7 +101,7 @@ def _execute(
     if the nth-order derivative is requested. Do not set this argument unless you
     understand the consequences!
     """
-    with qml.tape.Unwrap(*tapes):
+    with qml.tape.Unwrap(*tapes, set_trainable=False):

```

Comment:
How does this help recursive calls correctly determine the trainable parameters? Is it avoiding the trainable parameters altogether? But surely we need the trainable params at some point otherwise the bug wouldn't have been caught..

Reply:
Ok I think this makes sense in `pennylane/interfaces/batch/tensorflow.py`

Reply:
Yep, the same is happening in all three places - determining trainable parameters is happening before execution now!